### PR TITLE
Influxdb client: don't fill db.statement for create/drop database and write operations

### PR DESCRIPTION
**Pull request by laurit**
Fill `db.statement` only when user provides a query. Currently when user calls `createDatabase` and `deleteDatabase` we provide the same statement as what these methods execute. As the statement these methods execute is slightly different between influxdb client versions we may not always provide the correct statement. As far as I can tell `write` isn't really a query so filling `db.statement` doesn't make sense.
Adding okhttp instrumentation is needed because it provides an `IgnoredTypesConfigurer` that prevents a context leak. Previously somehow calling `influxDbImpl.version()`, that executes a request with okhttp, somehow avoided the context leak.

### CODE REVIEW

**Comment by steverao** [comment]
> Fill `db.statement` only when user provides a query. Currently when user calls `createDatabase` and `deleteDatabase` we provide the same statement as what these methods execute. As far as I can tell `write` isn't really a query so filling `db.statement` doesn't make sense. 

I think the meaning of query in `db.statement` is very general concept. It not only contains some query operation, such as `SELECT * FROM wuser_table where username = ?`, it also can be represent to other modify operations to database, such as `SET mykey "WuValue"`. Just like description in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/db.md#db-attributes, the `db.statement` replaced by `db.query.text` in latest semantic convention. But I think your thought is ok, we can don't provide `db.statement` for situations where users don't fill specific SQL.

> As the statement these methods execute is slightly different between influxdb client versions we may not always provide the correct statement. 

I don't understand here clearly. You think the execution is vary in different client version. Are there some situations that we have not tested?

> Adding okhttp instrumentation is needed because it provides an IgnoredTypesConfigurer that prevents a context leak. Previously somehow calling influxDbImpl.version(), that executes a request with okhttp, somehow avoided the context leak.

Sorry, for this problem, can you explain more? For example, which situation we need to avoid a context leak like it? or is there any detail context for it?

**Comment by laurit** [comment]
> I think the meaning of query in `db.statement` is very general concept. It not only contains some query operation, such as `SELECT * FROM wuser_table where username = ?`, it also can be represent to other modify operations to database, such as `SET mykey "WuValue"`. Just like description in https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/db.md#db-attributes, the `db.statement` replaced by `db.query.text` in latest semantic convention. But I think your thought is ok, we can don't provide `db.statement` for situations where users don't fill specific SQL.

In my opinion we should not fill `db.statement` with arbitrary text. Statement should resemble the actual query that was executed in the query language that the database supports. As far as I can tell `write` isn't a valid statement that could be executed in the database. As for create/drop if we can't capture the statement that was actually executed we don't need to synthesize the statement as it might not be correct.

>> As the statement these methods execute is slightly different between influxdb client versions we may not always provide the correct statement.

> I don't understand here clearly. You think the execution is vary in different client version. Are there some situations that we have not tested?

Have a look at https://github.com/influxdata/influxdb-java/blob/5f927b9b1718549f102a3de21bbdd5af5b248614/src/main/java/org/influxdb/impl/InfluxDBImpl.java#L224 here it conditionally executes either `CREATE DATABASE` or `CREATE DATABASE IF NOT EXISTS`
in https://github.com/influxdata/influxdb-java/blob/91d0f09f0c19bf5142d4c2e246155b80fc3f8ade/src/main/java/org/influxdb/impl/InfluxDBImpl.java#L690 it always executes `CREATE DATABASE`
If we want to fill the statement here we should rework the instrumentation to capture what was really executed instead of guessing.

>> Adding okhttp instrumentation is needed because it provides an IgnoredTypesConfigurer that prevents a context leak. Previously somehow calling influxDbImpl.version(), that executes a request with okhttp, somehow avoided the context leak.

> Sorry, for this problem, can you explain more? For example, which situation we need to avoid a context leak like it? or is there any detail context for it?

Comment out https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/okhttp/okhttp-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/okhttp/v3_0/OkHttp3IgnoredTypesConfigurer.java and try running okhttp tests.

**Comment by laurit** [comment]
> Do you mean we detected there are context leaks in InfluxDB by test harness? or we are just in order to avoid possible context leaks by adding okhttp in advance here?

The tests fail because of the leaked context without the okhttp instrumentation.

File: instrumentation/influxdb-2.4/javaagent/build.gradle.kts [hunk 1]
```diff
@@ -17,6 +17,8 @@ dependencies {
   compileOnly("com.google.auto.value:auto-value-annotations")
   annotationProcessor("com.google.auto.value:auto-value")
 
+  testInstrumentation(project(":instrumentation:okhttp:okhttp-3.0:javaagent"))
```
Comment (steverao) [inline instrumentation/influxdb-2.4/javaagent/build.gradle.kts:20]: Maybe is it  better to describe the context leak problem by adding necessary comment here?

**Review by steverao — APPROVED** [review]

File: instrumentation/influxdb-2.4/javaagent/build.gradle.kts (hunk 1 above)
Comment (laurit) [reply instrumentation/influxdb-2.4/javaagent/build.gradle.kts:20]: It is already described further down where the instrumentation is disabled.

Comment (steverao) [reply instrumentation/influxdb-2.4/javaagent/build.gradle.kts:20]: > It is already described further down where the instrumentation is disabled.

Yes, I noticed it before, but I was worried that someone might not have noticed the bottom comments.😅

**Review by trask — APPROVED** [review]

**Pull request merged**